src/actions/instantSale.ts

```diff
@@ -47,59 +51,57 @@ export const instantSale = async ({
     safetyDepositBox.pubkey,
   );
   const {
-    data: { winningConfigType },
+    data: { winningConfigType, participationConfig },
   } = await SafetyDepositConfig.load(connection, safetyDepositConfigPDA);
   ////
 
-  const {
-    txId: placeBidTxId,
-    bidderPotToken,
-    bidderMeta,
-  } = await placeBid({
+  const { txId: placeBidTxId, bidderPotToken } = await placeBid({
     connection,
     wallet,
     amount: instantSalePrice,
     auction,
+    // wait for all accounts to be created
+    commitment: 'finalized',
   });
   txIds.push(placeBidTxId);
 
-  // workaround to wait for bidderMeta to be created
-  await retry(async (bail) => {
-    await Account.getInfo(connection, bidderMeta);
-  });
+  const {
+    data: { bidState },
+  } = await Auction.load(connection, auction);
+  const winIndex = bidState.getWinnerIndex(wallet.publicKey.toBase58());
+  const hasWinner = winIndex !== null;
 
-  // NOTE: it's divided into 3 transactions since transaction size is restricted
-  switch (winningConfigType) {
-    case WinningConfigType.FullRightsTransfer:
-      const { txId: redeemFullRightsTransferBidTxId } = await redeemFullRightsTransferBid({
-        connection,
-        wallet,
-        store,
-        auction,
-      });
-      txIds.push(redeemFullRightsTransferBidTxId);
-      break;
-    case WinningConfigType.PrintingV2:
-      const { txId: redeemPrintingV2BidTxId } = await redeemPrintingV2Bid({
-        connection,
-        wallet,
-        store,
-        auction,
-      });
-      txIds.push(redeemPrintingV2BidTxId);
-      break;
-    default:
+  // NOTE: it's divided into several transactions since transaction size is restricted
+  if (hasWinner) {
+    if (winningConfigType === WinningConfigType.FullRightsTransfer) {
+      const { txId } = await redeemFullRightsTransferBid({ connection, wallet, store, auction });
+      txIds.push(txId);
+    } else if (winningConfigType === WinningConfigType.PrintingV2) {
```

Comment:
I feel deja vu, please use switch

Reply:
It's not necessary here since we have different conditions